pkg/runner/operations/delete/delete.go

```diff
@@ -29,50 +29,65 @@ func New(client client.Client, obj ctrlclient.Object, namespacer namespacer.Name
 	}
 }
 
-func (o *operation) Exec(ctx context.Context) (_err error) {
+func (o *operation) Exec(ctx context.Context) (err error) {
 	logger := logging.FromContext(ctx).WithResource(o.obj)
 	defer func() {
-		if _err == nil {
-			logger.Log(logging.Delete, logging.DoneStatus, color.BoldGreen)
+		if err != nil {
+			logger.Log(logging.Delete, logging.ErrorStatus, color.BoldRed, logging.ErrSection(err))
 		} else {
-			logger.Log(logging.Delete, logging.ErrorStatus, color.BoldRed, logging.ErrSection(_err))
+			logger.Log(logging.Delete, logging.DoneStatus, color.BoldGreen)
 		}
 	}()
+
 	if o.namespacer != nil {
-		if err := o.namespacer.Apply(o.obj); err != nil {
+		if err = o.namespacer.Apply(o.obj); err != nil {
 			return err
 		}
 	}
+
 	logger.Log(logging.Delete, logging.RunStatus, color.BoldFgCyan)
-	candidates, _err := internal.Read(ctx, o.obj, o.client)
-	if _err != nil {
-		if errors.IsNotFound(_err) {
+	return o.deleteResource(ctx, logger)
+}
+
+func (o *operation) deleteResource(ctx context.Context, logger logging.Logger) error {
+	candidates, err := internal.Read(ctx, o.obj, o.client)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
 			return nil
 		}
-		return _err
+		return err
 	}
+
 	for i := range candidates {
```

Comment:
it would be better to call all delete then wait for all deletions.

Reply:
right it would reduce the wait time for the deletion